metadata_backend/api/middlewares.py

```diff
@@ -50,6 +53,78 @@ async def http_error_handler(req: Request, handler: Callable) -> Response:
     return http_error_handler
 
 
+def jwt_middleware() -> Callable:
+    """Middleware for validating and authenticating JSON web token.
+
+    :param req: A request instance
+    :param handler: A request handler
+    :raises: HTTP Exception with status code 401 or 403
+    :returns: Successful requests unaffected
+    """
+    @middleware
+    async def jwt_authentication(req: Request, handler: Callable) -> Response:
+        if 'Authorization' in req.headers:
+            # Check token exists
+            try:
+                scheme, token = req.headers.get('Authorization').split(' ')
+                LOG.info('Auth token received.')
+            except Exception as err:
+                raise web.HTTPUnauthorized(reason=str(err))
+
+            # Check token has proper scheme and was provided.
+            if not re.match('Bearer', scheme):
+                raise web.HTTPUnauthorized(reason="Invalid token scheme, "
+                                                  "Bearer required.")
+            if token is None:
+                raise web.HTTPUnauthorized(reason='Token cannot be empty.')
+
+            # JWK and JWTClaims parameters for decoding
+            key = environ.get('PUBLIC_KEY', None)
+            # TODO: get Oauth2 public key
+            # TODO: verify audience claim
+
+            # Include claims that are required to be present
+            # in the payload of the token
+            claims_options = {
+                "iss": {
+                    "essential": True,
+                    # TODO: add the proper issuer URLs here
+                    "values": ["haka_iss", "elixir_iss"]
+                },
+                "exp": {
+                    "essential": True
+                }
+            }
+
+            # Decode and validate token
+            try:
+                claims = jwt.decode(token, key, claims_options=claims_options)
+                claims.validate()
+                LOG.info('Auth token decoded and validated.')
+
+                # TODO: Retrieve GA4GH passports and process into permissions
```

Comment:
this is done here: https://github.com/CSCfi/beacon-python/blob/dev/beacon_api/permissions/ga4gh.py

Reply:
Added the issue #96, which also mentions this.